app/src/org/commcare/android/nsd/NsdTools.java

```diff
@@ -0,0 +1,202 @@
+package org.commcare.android.nsd;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.net.nsd.NsdManager;
+import android.net.nsd.NsdServiceInfo;
+import android.os.Build;
+import android.util.Log;
+
+import java.net.InetAddress;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Manages relevant hooks for communicating with devices on the local network
+ * like the CommCareHub app. Provides a place for registering listeners independent
+ * from the app lifecycle and translating that into actionable hooks.
+ *
+ * Created by ctsims on 2/19/2016.
+ */
+public class NsdTools {
```

Comment:
`NsdTools` -> `NSDLocator` or something that mentions the fact that this class discovers local network services.